### PR TITLE
app-emulation/docker: Include fish-shell completion file

### PR DESCRIPTION
**Pull request by Asgavar**
Bash and zsh were already included, but upstream supports all three of them.

### CODE REVIEW

File: app-emulation/docker/docker-18.03.1.ebuild [hunk 1]
```diff
@@ -282,6 +282,8 @@ src_install() {
 	doman man/man*/*
 
 	dobashcomp contrib/completion/bash/*
+	insinto /usr/share/fish/completions/
```
Comment (gyakovlev) [inline app-emulation/docker/docker-18.03.1.ebuild]: please install 3rd party fish completions to `/usr/share/fish/vendor_completions.d/`
that is intended dir for packages to install completions to.
I should probably document it somewhere )

File: app-emulation/docker/docker-18.05.0.ebuild [hunk 2]
```diff
@@ -282,6 +282,8 @@ src_install() {
 	doman man/man*/*
 
 	dobashcomp contrib/completion/bash/*
+	insinto /usr/share/fish/completions/
```
Comment (gyakovlev) [inline app-emulation/docker/docker-18.05.0.ebuild]: same

File: app-emulation/docker/docker-9999.ebuild [hunk 3]
```diff
@@ -261,6 +261,9 @@ src_install() {
 	insinto /usr/share/zsh/site-functions
 	doins contrib/completion/zsh/_*
 
+	insinto /usr/share/fish/completions/
```
Comment (gyakovlev) [inline app-emulation/docker/docker-9999.ebuild]: ditto

**Comment by Asgavar** [comment]
Amended! Now they go where they are supposed to.

**Comment by Asgavar** [comment]
Okay, now they are split in two and 18.03.1 stays untouched. Not sure about mentioning concrete versions in commits though, so feel free to share your thoughts on this.

**Comment by gyakovlev** [comment]
mentioning version is not strictly required but help then checking commit log to see what happened.

first/summary line of commit messages is too long, should 69 or less characters.

you can get away with just 
`app-emulation/docker: install fish completion`
^ for 9999
`app-emulation/docker: revbump to 18.05.0-r1, install fish completion`


that will be informative and just under 69 character limit =)

Thanks for figuring it out,
I maintain fish ebuild and use docker, but always installed 3rd party docker completion without realizing they ship it in contrib now)

**Comment by Asgavar** [comment]
Everything should be perfect now :>

**Comment by gyakovlev** [comment]
```diff
--- docker-18.05.0.ebuild       2018-06-16 17:34:31.989281989 -0700
+++ docker-18.05.0-r1.ebuild    2018-06-19 15:06:56.145043414 -0700
@@ -282,6 +282,8 @@
        doman man/man*/*
 
        dobashcomp contrib/completion/bash/*
+       insinto /usr/share/fish/vendor_completions.d/
+       doins contrib/completion/fish/docker.fish
        insinto /usr/share/zsh/site-functions
        doins contrib/completion/zsh/_*
        popd || die # components/cli
```

**Comment by gyakovlev** [comment]
pushed, thanks!

**Pull request closed without merging**